test/PushNotificationTests/UnpackagedSelfContained.cpp

```diff
@@ -0,0 +1,214 @@
+            {
+                // Cleanup previous installations. Need this to remove any manual installations outside of running this tests.
+                TP::RemovePackage_PushNotificationsLongRunningTask();
+                TP::RemovePackage_DynamicDependencyLifetimeManager();
+                TP::RemovePackage_DynamicDependencyDataStore();
+                TP::RemovePackage_WindowsAppRuntimeFramework();
+
+                TP::AddPackage_WindowsAppRuntimeFramework();       // Installs WARfwk
+                TP::AddPackage_DynamicDependencyDataStore();       // Installs WARmain
+                TP::AddPackage_DynamicDependencyLifetimeManager(); // Installs WARddlm
+                TP::AddPackage_PushNotificationsLongRunningTask(); // Installs the PushNotifications long running task.
+            }
+            catch (...)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            try
+            {
+
+                // Remove in reverse order to avoid conflicts between inter-dependent packages.
+                TP::RemovePackage_PushNotificationsLongRunningTask();
+                TP::RemovePackage_DynamicDependencyLifetimeManager();
+                TP::RemovePackage_DynamicDependencyDataStore();
+                TP::RemovePackage_WindowsAppRuntimeFramework();
+            }
+            catch (...)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppRuntimeFramework());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_PushNotificationsLongRunningTask());
+
+            VERIFY_ARE_EQUAL(S_OK, MddBootstrapTestInitialize(Test::Packages::DynamicDependencyLifetimeManager::c_PackageNamePrefix, Test::Packages::DynamicDependencyLifetimeManager::c_PackagePublisherId));
+
+            // Major.Minor version, MinVersion=0 to find any framework package for this major.minor version
+            const UINT32 c_Version_MajorMinor{ Test::Packages::DynamicDependencyLifetimeManager::c_Version_MajorMinor };
+            const PACKAGE_VERSION c_minVersion{};
+            VERIFY_ARE_EQUAL(S_OK, MddBootstrapInitialize(c_Version_MajorMinor, nullptr, c_minVersion));
+
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            // Need to keep each TEST_METHOD in a clean state in the LRP.
+            try
+            {
+                PushNotificationManager::Default().UnregisterAll();
+            }
+            catch (...)
+            {
+                // We want to unregister regardless of pass/fail to clean the state.
+            }
+            MddBootstrapShutdown();
```

Comment:
Does this return HRESULT or is it just void?

Reply:
Void

Reply:
Void